test/core/ext/filters/event_engine_client_channel_resolver/resolver_ops.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package event_engine_client_channel_resolver;
+
+message ResolverArgs {
+    bool request_service_config = 2;
+    bool enable_srv_queries = 3;
+}
+
+// Some error. The details are unimportant.
+message Error {}
+
+
+message ResolvedAddress {
+    uint32 port = 1;
+}
+message ResolvedAddresses {
+    repeated ResolvedAddress addresses = 1;
+}
+
+message SRVRecord {
+    string host = 1;
+    uint32 port = 2;
+    uint32 priority = 3;
+    uint32 weight = 4;
+}
+message SRVRecords {
+    repeated SRVRecord srv_records = 1;
+}
+
+enum TXTRecordType {
+    TXT_VALID = 0;
+    TXT_INVALID = 1;
+}
+message TXTRecords {
+    repeated TXTRecordType txt_records = 1;
```

Comment:
Shouldn't this include the actual text of the TXT record to return?

Reply:
This fuzzer reuses 2 cases which are hardcoded as base cases of the fuzzer: one TXT record that is parseable as a valid gRPC config, and the other "invalid" gRPC config. We could add more cases or have more content randomized, but I definitely wanted the fuzzer to test the happy path as well, and randomly generating a correct config is more complicated. If we wanted to fuzz the parser (maybe separately), we could.

There were problems early on where valid and invalid TXT responses would not mix, for example. This fuzzer exercises that.

Reply:
I'm less interested in the service config parsing piece, since I think we can fuzz that separately (although I don't know if we are actually doing so -- we should look at that).  I'm more interested in testing the code in the client channel resolver that chooses which TXT entry to use, as per the changes we made to the EE TXT lookup API in #33030.

Reply:
This PR now fuzzes the service config as well, so the parser is going to get some exercise too.

> I'm more interested in testing the code in the client channel resolver that chooses which TXT entry to use, as per the changes we made to the EE TXT lookup API in https://github.com/grpc/grpc/pull/33030.

That was already being fuzzed ... it was exercised without respect to whether it was doing "the right thing", but the fuzzer generated a series of valid & invalid records, and the resolver had to find one with a config prefix.

In any case, I've made this more robust, including options for non-config TXT records, non-parseable `grpc_config=` TXT records, and fuzzer-generated ServiceConfig JSON.